[Pillar] Add optional link (#290)
primer/brand: apps/docs/content/components/Pillar/index.mdx

When presenting more than four pillars, we recommend displaying them in multiple rows. You can use the [Grid](/components/Grid) or the [Stack](/components/Stack) component to achieve this. We recommend using a grid with 3 or 4 columns and balancing the number of pillars on each row.

![An image showing a group of numerous Pillars stacked in a 3 column grid.](https://github.com/primer/brand/assets/912236/1d031ae9-bf4e-4e19-a7b7-c8cc0b7e5940)

### Pillar and Card

> Pillar and Card components might lead to ambiguity when deciding which one to use, as both could be used to provide pieces of related and condensed information. Some of the differences between them are:

Reviewer: Suggestion to remove the assumption around ambiguity.

```suggestion
Pillar and Card are both components that are used to present related and/or condensed information. They have distinct use-cases however, with the main differences between them being:
```

#### Interaction

> Pillars are used mainly for plain content and might include an optional link for extended information. On the other hand, Cards are primarily navigational elements.

Reviewer: Could we elaborate on what plain here means? Do you mean non-interactive? 

```suggestion
Pillars are used mainly for non-interactive content and might include an optional link for extended information. On the other hand, Cards are primarily navigational elements.
```

#### Hierarchy

> Cards should be used for noteworthy information to draw users attention. Pillars, in the other hand, should be used for common, less noticeable information compared to Cards.

Reviewer: Alternative wording suggestion. I wasn't sure what 'common' meant here, but think lower priority could be a good alternative? LMK if i'm off-base here.

```suggestion
Cards should be used for noteworthy information to immediately draw user attention. Pillars, on the other hand, should be used for lower priority information compared to Cards.
```

#### Group size

> Cards should be used in small groups (3, 4 or 5 items maximum) whereas Pillars might be used on both small and big groups (more than 5 items). The more items a users see's within a group, the less noticeable they become as unique elements.

Reviewer: Would you ever want to use cards in a group of 5? 

```suggestion
Cards should be used in small groups (3 or 4 maximum), whereas Pillars can be used in larger or smaller groups (5+). The more items a users sees within a group, the less noticeable they become as unique elements.
```

Reviewer: Or slight alternative to the last line..

```suggestion
Cards should be used in small groups (3 or 4 maximum), whereas Pillars can be used in larger or smaller groups (5+). As users encounter an increasing number of items within a group, the distinctiveness of each individual element diminishes.
```

Author: Love the last suggestion 👍

> Cards should be used in small groups (3, 4 or 5 items maximum) whereas Pillars might be used on both small and big groups (more than 5 items). The more items a users see's within a group, the less noticeable they become as unique elements.

Reviewer: Also a separate question around this bit:

> Pillars might be used on both **small** and big groups

Would you ever want to use Pillars in less than three config? Not sure if we should encourage that as there are better ways to present sparse infomation. Should this guidance be focussed on recommending Pillars for larger quantities of grouped information?

![An image showing the Pillar anatomy with an icon, a heading and a description in a vertical order.](https://github.com/primer/brand/assets/912236/1d031ae9-bf4e-4e19-a7b7-c8cc0b7e5940)

### Pillar and Card

Pillar and Card components might lead to ambiguity when deciding which one to use, as both could be used to provide pieces of related and condensed information. Some of the differences between them are:

#### Interaction

> Cards should be used as navigational items to lead to the extend of the information, the link is always required. Pillars could be used for just plain, non-navigational content or include an optional link and lead to extra information.

Author: ```suggestion
Pillars are used mainly for plain content and might include an option link for extended information. On the other hands, Cards are primarily navigational elements.
```
